Userland/Libraries/LibCompress/LZW.h

```diff
@@ -17,23 +17,85 @@
 
 namespace Compress::LZW {
 
-template<InputBitStream InputStream>
-class Decoder {
-private:
-    static constexpr int max_code_size = 12;
+namespace Details {
 
+class Base {
```

Comment:
`Base` isn't a great name. Maybe just `Codec`?

Reply:
Following the naming scheme from other compressors/decompressors (which I'd like to keep consistent rather than inventing something new with additional namespaces), these would have been `LzwDecoder`, `LzwEncoder`, and `LzwState` (or whatever the most appropriate name is) respectively.

My full review will have to wait until a later date, but I figured I'd comment this already since we are touching on the topic.

Reply:
Ok, I brought back `LZWDecoder` and named the base class `LZWState`